x-pack/plugins/security_solution/public/resolver/store/data/node_events_in_category_model.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { PanelViewAndParameters, NodeEventsInCategoryState } from '../../types';
+
+/**
+ * True if `nodeEventsInCategory` contains data that is relevant to `panelViewAndParameters`.
+ * This is used by the reducer to enforce that the nodeEventsInCategory state is always valid.
+ */
+export function isValid(
+  nodeEventsInCategory: NodeEventsInCategoryState,
+  panelViewAndParameters: PanelViewAndParameters
+): boolean {
+  return (
+    panelViewAndParameters.panelView === 'nodeEventsInCategory' &&
+    panelViewAndParameters.panelParameters.nodeID === nodeEventsInCategory.nodeID &&
+    panelViewAndParameters.panelParameters.eventCategory === nodeEventsInCategory.eventCategory
+  );
+}
+
+/**
+ * Return an updated `NodeEventsInCategoryState` that has data from `first` and `second`. The `cursor` from `second` is used.
+ * Returns undefined if `first` and `second` don't contain data form the same set.
+ */
+export function updatedWith(
```

Comment:
❔ At first read, this seems like something that should be in a selector or a reducer rather than a model, but 🤷

Reply:
This is only used from the data reducer. The module is under `store/data`.